lm-feedCore-iOS/lm-feedCore-iOS/Source/Classes/LMFeedCore.swift

```diff
@@ -64,28 +65,27 @@ public class LMFeedCore {
         }
     }
     
-    func showFeed(accessToken: String, refreshToken: String, handler: LMFeedCoreCallback?, completionHandler: ((Result<Void, LMFeedError>) -> Void)?) {
-        self.coreCallback = handler
-        
+    func showFeed(accessToken: String, refreshToken: String, completionHandler: ((Result<Void, LMFeedError>) -> Void)?) {
         let request = ValidateUserRequest
             .builder()
             .accessToken(accessToken)
             .refreshToken(refreshToken)
             .build()
         
-        LMFeedClient.shared.validateUser(request) { response in
+        LMFeedClient.shared.validateUser(request) { [weak self] response in
             guard response.success else {
                 completionHandler?(.failure(.apiInitializationFailed(error: response.errorMessage)))
                 return
             }
             
             if response.data?.appAccess == false {
-                self.logout(refreshToken, deviceId: UIDevice.current.identifierForVendor?.uuidString ?? "")
+                self?.logout(refreshToken, deviceId: UIDevice.current.identifierForVendor?.uuidString ?? "")
                 completionHandler?(.failure(.appAccessFalse))
                 return
             }
             
             Self.isInitialized = true
+            self?.fetchCommunityConfiguration()
```

Comment:
this should be a blocker function, we should only return completionHandler after the response from this function. Confirm this once with @ishaan1607

Reply:
why should it be a blocker function, and what will be the consequences of this failing ?